src/transformers/models/patchtst/configuration_patchtst.py

```diff
@@ -0,0 +1,269 @@
+        patch_length: int = 1,
+        stride: int = 1,
+        # Transformer architecture configuration
+        encoder_layers: int = 3,
+        d_model: int = 64,
+        encoder_attention_heads: int = 4,
+        shared_embedding: bool = True,
+        channel_attention: bool = False,
+        encoder_ffn_dim: int = 256,
+        norm: str = "BatchNorm",
+        attention_dropout: float = 0.0,
+        dropout: float = 0.0,
+        positional_dropout: float = 0.0,
+        dropout_path: float = 0.0,
+        ff_dropout: float = 0.0,
+        bias: bool = True,
+        activation_function: str = "gelu",
+        pre_norm: bool = False,
+        positional_encoding: str = "sincos",
+        learn_pe: bool = False,
+        use_cls_token: bool = False,
+        init_std: float = 0.02,
+        shared_projection: bool = True,
+        seed_number: Optional[int] = None,
+        scaling: Optional[Union[str, bool]] = "mean",
+        # mask pretraining
+        mask_input: Optional[bool] = None,
+        mask_type: str = "random",
+        mask_ratio: float = 0.5,
+        mask_patches: List[int] = [2, 3],
+        mask_patch_ratios: List[int] = [1, 1],
+        channel_consistent_masking: bool = False,
+        unmasked_channel_indices: Optional[List[int]] = None,
+        mask_value=0,
+        # head
+        pooling: str = "mean",
+        num_classes: int = 1,
+        head_dropout: float = 0.0,
+        prediction_length: int = 24,
+        num_output_channels: int = 1,
+        prediction_range: List = None,
+        # distribution head
+        num_parallel_samples: int = 100,
+        **kwargs,
+    ):
+        # time series specific configuration
+        self.context_length = context_length
+        self.num_input_channels = num_input_channels  # n_vars
+        self.loss = loss
+        self.distribution_output = distribution_output
+        self.num_parallel_samples = num_parallel_samples
+
+        # Transformer architecture configuration
+        self.d_model = d_model
+        self.encoder_attention_heads = encoder_attention_heads
+        self.encoder_ffn_dim = encoder_ffn_dim
+        self.encoder_layers = encoder_layers
+        self.dropout = dropout
+        self.attention_dropout = attention_dropout
+        self.shared_embedding = shared_embedding
+        self.channel_attention = channel_attention
+        self.norm = norm
+        self.positional_dropout = positional_dropout
+        self.dropout_path = dropout_path
+        self.ff_dropout = ff_dropout
+        self.bias = bias
+        self.activation_function = activation_function
+        self.pre_norm = pre_norm
+        self.positional_encoding = positional_encoding
+        self.learn_pe = learn_pe
+        self.use_cls_token = use_cls_token
+        self.init_std = init_std
+        self.scaling = scaling
+
+        # PatchTST
+        self.patch_length = patch_length
+        self.stride = stride
+        self.num_patches = self._num_patches()
```

Comment:
for ViT, we only calculate the number of patches in the model, not in the config